cpp-terminal/key.cpp

```diff
@@ -13,63 +61,66 @@ std::string Term::Key::str() const
   }
 }
 
-Term::Key::Key(const Term::Key::Value& value) : m_value(value) {}
-
-Term::Key::operator Term::Key::Value() const { return m_value; }
-
 bool Term::Key::isASCII() const
 {
-  if(m_value >= NUL && m_value <= DEL) return true;
+  if(m_value >= Null && m_value <= Del) return true;
   else
     return false;
 }
 
 bool Term::Key::isExtendedASCII() const
 {
-  if(m_value >= NUL && m_value <= 255) return true;
+  if(m_value >= Null && m_value <= 255) return true;
   else
     return false;
 }
 
-bool Term::Key::isCTRL() const
+bool Term::Key::hasCtrl() const
 {
   // Need to suppress the TAB etc...
-  if(iscntrl() && m_value != Key::BACKSPACE && m_value != Key::TAB && m_value != ESC && m_value != ENTER && m_value != DEL) return true;
+  if((iscntrl() || hasCtrlAll()) && m_value != Key::Backspace && m_value != Key::Tab && m_value != Esc && m_value != Enter && m_value != Del) return true;
   else
     return false;
 }
 
-bool Term::Key::isALT() const
+bool Term::Key::hasCtrlAll() const
 {
-  if(m_value >= Key::ALT && Key::ALT < (Key::ALT << 1)) return true;
+  if((iscntrl() || ((m_value & Term::MetaKey::Value::Ctrl) == Term::MetaKey::Value::Ctrl))) return true;
+  else
+    return false;
+}
+
+bool Term::Key::hasAlt() const
+{
+  if((m_value & Term::MetaKey::Value::Alt) == Term::MetaKey::Value::Alt) return true;
   else
     return false;
 }
 
 bool Term::Key::empty() const
 {
-  if(m_value == NO_KEY) return true;
+  if(m_value == NoKey) return true;
   else
     return false;
 }
 
 bool Term::Key::iscntrl() const
 {
-  if((m_value >= NUL && m_value <= CTRL_UNDERSCORE) || m_value == DEL) return true;
+  if((m_value >= Null && m_value <= Ctrl_Underscore) || m_value == Del) return true;
   else
     return false;
 }
 
 bool Term::Key::isblank() const
 {
-  if(m_value == TAB || m_value == SPACE) return true;
+  if(m_value == Tab || m_value == Space) return true;
```

Comment:
Why have an if, just return the expression?

You did that a few times, i stop here and won't com.ent ever one.

Reply:
Yes thx, I wonder if there is some impact. Nowdays compilers are very efficient 1nd maybe detect my stupid logic but I agree for the reader it make life easier just to return and not do the if things. Will change them

Reply:
Yes, I bet compilers do that. Maybe not in the C++ syntax optimizer but the assembler optimizer will probably optimize it. So I won't think it has an impact, it is mostly coding style, but I do think that it may reduce human errors if there are less lines that can be wrong.